salt/auth/pam.py

```diff
@@ -228,7 +228,7 @@ def authenticate(username, password):
     env["SALT_PAM_PASSWORD"] = password
     env["SALT_PAM_SERVICE"] = __opts__.get("auth.pam.service", "login")
     env["SALT_PAM_ENCODING"] = __salt_system_encoding__
-    pyexe = pathlib.Path(__opts__.get("auth.pam.python", "/usr/bin/python3")).resolve()
+    pyexe = pathlib.Path(__opts__.get("auth.pam.python", sys.executable)).resolve()
```

Comment:
I agree this should be more robust but we can not simply `sys.executable`. The intention here is to use the system's python interpreter in onedir environments. This change will likely break that intention.

We may want to consider adding a configuration variable where a user can define their system python location.

Reply:
Thanks for clarifying.

Since there is already a configuration variable for this: `auth.pam.python`, I've just enhanced the behavior to use system's python interpreter and just fallback to `sys.executable` if it does not exist, to make this more reliable in case of Python 3 interpreter not available. What do you think?

BTW, I also noticed this issue: https://github.com/saltstack/salt/issues/66035 which is reporting this exact situation. I added corresponding changelog entry and links to this PR.